test/unit_test/components/movement_sensor_test.dart

```diff
@@ -0,0 +1,285 @@
+    });
+
+    group('MovementSensor Service Tests', () {
+      test('accuracy', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getAccuracy(GetAccuracyRequest(name: name));
+        expect(result.accuracy, {'accuracy': 0});
+      });
+
+      test('angularVelocity', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getAngularVelocity(GetAngularVelocityRequest(name: name));
+        expect(result.angularVelocity.x, 0);
+      });
+
+      test('compassHeading', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getCompassHeading(GetCompassHeadingRequest(name: name));
+        expect(result.value, 0);
+      });
+
+      test('linearAcceleration', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getLinearAcceleration(GetLinearAccelerationRequest(name: name));
+        expect(result.linearAcceleration.x, 0);
+      });
+
+      test('linearVelocity', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getLinearVelocity(GetLinearVelocityRequest(name: name));
+        expect(result.linearVelocity.x, 0);
+      });
+
+      test('orientation', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getOrientation(GetOrientationRequest(name: name));
+        expect(result.orientation.oX, 0);
+      });
+
+      test('position', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getPosition(GetPositionRequest(name: name));
+        expect(result.altitudeM, 0);
+      });
+
+      test('properties', () async {
+        final client = MovementSensorServiceClient(channel);
+        final result = await client.getProperties(GetPropertiesRequest(name: name));
+        expect(result.angularVelocitySupported, false);
+      });
+
+      test('doCommand', () async {
+        final cmd = {'foo': 'bar'};
+
+        final client = MovementSensorServiceClient(channel);
+        final resp = await client.doCommand(DoCommandRequest(name: name, command: cmd.toStruct()));
+        expect(resp.result.toMap()['command'], cmd);
+      });
+
+      test('extra', () async {
+        expect(movementSensor.extra, null);
+
+        final client = MovementSensorServiceClient(channel);
+        await client.getAccuracy(GetAccuracyRequest(name: name, extra: {'foo': 'bar'}.toStruct()));
+        expect(movementSensor.extra, {'foo': 'bar'});
+      });
+    });
+    group('MovementSensor Client Tests', () {
+      test('accuracy', () async {
+        final client = MovementSensorClient(name, channel);
+        final result = await client.accuracy();
+        expect(result, {'accuracy': 0});
+      });
+
+      test('angularVelocity', () async {
```

Comment:
how is this angularVelocity test different from the other above?

Reply:
there are three of these: fake, client, and server tests for each function

Reply:
yes, exactly, so you'll see the same test 3 name 3 times in this file, but it will be testing calling the method from the fakeResource we built for testing purposes, the client and the server. so if a test starts to fail we have visibility into where exactly it is failing.